<a href="https://colab.research.google.com/github/bioqu/bioqu/blob/main/Copia_de_Clase_S08.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Problema 01**
Se encuentra que la concentración promedio de zinc que se obtiene a partir de una muestra de mediciones de zinc en 36 sitios diferentes es 
2.6 gramos por mililitro. Suponiendo que la desviación estándar de la población es 0.3, determine

a) El intervalo de confianza de 95% para la concentración media de zinc en el río. 

In [ ]:
import numpy as np
import scipy.stats as ss

In [ ]:
####  Llenar los datos y ejecutar --> click
alpha=0.05      # Nivel de significancia
n=36            # Tamaño de la muestra
media=2.6    # Media o promedio de la muestra
desv_poblacional=0.3   # Desviación estándar o típica de la población
#############
#############
#############
z_critico=ss.norm.ppf(1-alpha/2)
z_critico
Limite_inf=media-z_critico*desv_poblacional/np.sqrt(n)
print('Límite inferior:',Limite_inf)
Limite_sup=media+z_critico*desv_poblacional/np.sqrt(n)
print('Límite superior:',Limite_sup)
print('Intervalo de confianza:',[Limite_inf,Limite_sup])

Límite inferior: 2.502001800772997
Límite superior: 2.697998199227003
Intervalo de confianza: [2.502001800772997, 2.697998199227003]


**Resultado:** El intervalo de confianza de 95% para la concentración media de zinc en el río es de 2.50 a 2.70 gramos por mililitro.

b) El error que se comete al estimar un intervalo de confianza para la media con un nivel de significancia de 0.01.



c)¿Qué tan grande se requiere una muestra si queremos tener 95%
de confianza de que nuestra estimación de la media difiera por menos de 0.05?

**Problema 02** En una universidad se desea conocer la estatura (metros) promedio de sus estudiantes, para lo que se toma una muestra a 100 estudiantes, obteniendo los siguientes resultados:
\begin{array}{|c|c|}
\hline
Estatura  & Frecuencia\\
\hline
[1.52-1.60)&5\\
\hline
[1.60-1.68)&18\\
\hline
[1.68-1.76)&42\\
\hline
[1.76-1.84)&27\\
\hline
[1.84-1.92)&8\\
\hline
\end{array}


a) Determine el intervalo de confianza de 95% para la verdadera estatura promedio de los estudiantes.



b) Se piensa que la desviación estándar (típica) de las estaturas es mayor a 0.07 m. Realice una prueba de hipótesis y verifique. Considere un nivel de confianza del 95%.

c) Determine una prueba de hipótesis para la verdadera proporción de estudiantes cuya estatura supera 1.76 metros.

# Prueba de bondad de ajuste

El test de bondad de ajuste se utiliza para determinar si una colección de datos se ajusta a una
distribución teórica de una determinada población. Nos dice si la muestra disponible representa o
se ajusta de manera razonable a los datos que se esperan encontrar en la población.

La fórmula para el cálculo del estadı́stico chi cuadrado utilizado en el test de bondad de ajuste es 

$\chi^2_{prueba}=\Sigma_i \dfrac{(O_i-E_i)^2}{E_i} $

$O_i:   \ $Corresponde al valor observado.

$Ei: \ $Corresponde al valor esperado de la clase u observación i-ésima.

**Formulación de la hipótesis**

**H0:** Los datos se ajustan a una distribución dada.

**H1:** Los datos no se ajustan a una distribución dada.

**Valor crítico**: $\chi^2_{1-\alpha; k-1}$ donde $k$ es el número de clases

**P_valor:** $P\left(X>  \chi^2_{prueba}\right)$ 

**Función en python**:
ss.chisquare(observado,esperado)



**Problema 03** Las proporciones de tipos de sangre A, B, AB y O en la población de todos los caucasianos en Estados Unidos son 0.41, 0.10, 0.04 y 0.45, respectivamente. Para determinar si las proporciones poblacionales reales se ajustan o no se ajustan a este conjunto de probabilidades reportadas, se seleccionó una muestra de 200 estadounidenses y se registraron
sus fenotipos sanguíneos. 
\begin{array}{|c|c|c|c|c|}
\hline
Tipo & A  &  B &  AB &0\\
\hline
Observación  \ O_i & 89  &  18 &  18 & 81\\
\hline
\end{array}

Pruebe la bondad de ajuste de estas proporciones de fenotipo sanguíneas.


**Problema 04:**  La señora Patricia Mora es la gerente de marketing de un fabricante de tarjetas deportivas. Ella planea iniciar la venta de una serie de tarjetas con fotografı́as y estadı́sticas de juego de ex jugadores de las ligas mayores de béisbol. Uno de los problemas es la selección de ex jugadores. En una exhibición de tarjetas de béisbol en Southwyck Mall el pasado fin de semana, instaló un puesto y ofreció tarjetas seis jugadores miembros del
Salón de la Fama, así al final del dı́a vendió un total de 120 tarjetas. El número de tarjetas vendidas de cada jugador aparece en la siguiente tabla:


\begin{array}{|c|c|}
\hline
Jugador & Tarjetas \ \  vendidas\\
\hline
Tom \ \ Seaver & 13\\
\hline
Nolan  \ \ Ryan & 33\\
\hline
Ty  \ \  Cobb & 14\\
\hline
George  \ \  Brett & 7\\
\hline
Hank  \ \ Aaron & 36\\
\hline
Johnny  \ \  Bench & 17\\
\hline
Total & 120\\
\hline
\end{array}

¿La señora Mora puede concluir que las ventas no son iguales por cada jugador? 

Considere $\alpha = 0,05.$



#Test de independencia chi-cuadrado


La χ2 es una prueba de libre distribución (no paramétrica) que mide la diferencia o discrepancia
entre una distribución de frecuencias observadas y esperadas. Sirve para estudiar la independen-
cia de dos variables aleatorias.

**Formulación de la hipótesis**

**H0:** No hay relación entre las variables.

**H1:** Hay relación entre las variables.

**Estadístico de prueba** $\chi^2_{prueba}=\Sigma_i \Sigma_j\dfrac{(O_{ij}-E_{ij})^2}{E_{ij}} $


**Valor crítico**: $\chi^2_{1-\alpha; (a-1)*(b-1)}$ donde $a$ es el número de filas y $b$ el número de columnas.

**P_valor:** $P\left(X>  \chi^2_{prueba}\right)$ 

**Función en python**:
ss.chi2.contingency(observaciones)

**Problema 05:** Se encuestó a una muestra aleatoria de personas de todos los grupos socioeconómicos. De un listado de marcas de bebidas gaseosas, los encuestados debı́an elegir la de mayor preferencia (U, V o W). Los resultados con las frecuencias observadas de preferencias
se muestran a continuación:
\begin{array}{|c|c|c|c|}
\hline
Grupo  \ O_i / Marca & U  &  V &  W \\
\hline
ABC & 12  &  11 &  16 \\
\hline
D & 9  &  6 &  3 \\
\hline
E & 0  &  5 &  9 \\
\hline
\end{array}

Verificar si la elección de la marca de bebida gaseosa está condicionada al nivel socioeconómico.

Use una significancia del $5 \%$.
